tests/extensions/extra/footnote.html

```diff
@@ -1,5 +1,8 @@
 <p>This is the body with a footnote<sup id="fnref:1"><a class="footnote-ref" href="#fn:1" rel="footnote">1</a></sup> or two<sup id="fnref:2"><a class="footnote-ref" href="#fn:2" rel="footnote">2</a></sup> or more<sup id="fnref:3"><a class="footnote-ref" href="#fn:3" rel="footnote">3</a></sup> <sup id="fnref:4"><a class="footnote-ref" href="#fn:4" rel="footnote">4</a></sup> <sup id="fnref:5"><a class="footnote-ref" href="#fn:5" rel="footnote">5</a></sup>.</p>
 <p>Also a reference that does not exist[^6].</p>
+<p>Duplicate<sup id="fnref:a"><a class="footnote-ref" href="#fn:a" rel="footnote">6</a></sup> footnotes<sup id="fnref2:a"><a class="footnote-ref" href="#fn:a" rel="footnote">6</a></sup> test<sup id="fnref2:a"><a class="footnote-ref" href="#fn:a" rel="footnote">6</a></sup>.</p>
+<p>Duplicate<sup id="fnref:b"><a class="footnote-ref" href="#fn:b" rel="footnote">7</a></sup> footnotes<sup id="fnref2:b"><a class="footnote-ref" href="#fn:b" rel="footnote">7</a></sup> test<sup id="fnref2:b"><a class="footnote-ref" href="#fn:b" rel="footnote">7</a></sup>.</p>
```

Comment:
Footnote 6 is repeated three times on the first line (once with `id="fnref:a"` and twice with `id="fnref2:a"`.

Footnote 7 is repeated three times on the second line (once with `id="fnref:a"` and twice with `id="fnref2:a"`.

However, in the actual footnotes below, we have three backreference links to each line line as:

footnote 6: `href="#fnref:a"`, footnote 7: `href="#fnref2:a"` and footnote 8: `href="#fnref3:a" `
footnote 9: `href="#fnref:a"`, footnote 10: `href="#fnref2:a"` and footnote 11: `href="#fnref3:a" `

We have two problems. 

1. The footnote numbers don't match between the references and the backrefs. If we "jump back to footnote 8", then we should jump back to a footnote labeled "8" (i.e.; `<sup>8</sup>`)

2. There appears to be a bug in generating ids for the duplicates when there are more than two. Curiously, the backrefs point to the correct ids, but the references (after the first) are all assigned `"fnref2:a"` rather than counting up (`"fnref2:a", "fnref3:a", ...`).